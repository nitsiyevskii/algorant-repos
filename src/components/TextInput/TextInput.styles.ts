import { createStyles } from "src/utils/createStyles";

export const useStyles = createStyles((theme) => ({
  container: {
    flexDirection: "row",
    alignItems: "center",
    borderRadius: 10,
    borderWidth: 1,
    paddingHorizontal: 12,
    height: 48,
    backgroundColor: theme.colors.input.background,
    borderColor: theme.colors.input.border,
  },
  leftIconContainer: {
    marginRight: 8,
  },
  input: {
    flex: 1,
    height: "100%",
    paddingVertical: 0,
    color: theme.colors.input.text,
    fontSize: theme.fonts.size.md,
  },
  clearButton: {
    marginLeft: 8,
    padding: 4,
  },
}));

