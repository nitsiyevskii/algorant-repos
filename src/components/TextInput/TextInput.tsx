import React from 'react';
import {
  View,
  TextInput as RNTextInput,
  TouchableOpacity,
  TextInputProps as RNTextInputProps,
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { useTheme } from 'src/hooks/useTheme';
import { useStyles } from './TextInput.styles';

interface TextInputProps extends Omit<RNTextInputProps, 'onChange'> {
  value: string;
  onChangeText: (text: string) => void;
  placeholder?: string;
  leftIcon?: keyof typeof Ionicons.glyphMap;
  showClearButton?: boolean;
  disabled?: boolean;
}

export const TextInput: React.FC<TextInputProps> = ({
  value,
  onChangeText,
  placeholder = 'Search...',
  leftIcon = 'search',
  showClearButton = true,
  disabled = false,
  ...rest
}) => {
  const theme = useTheme();
  const styles = useStyles();

  const handleClear = () => {
    onChangeText('');
  };

  return (
    <View
      style={[
        styles.container,
        { opacity: disabled ? 0.6 : 1 },
      ]}
    >
      {leftIcon && (
        <View style={styles.leftIconContainer}>
          <Ionicons
            name={leftIcon}
            size={20}
            color={theme.colors.input.placeholder}
          />
        </View>
      )}

      <RNTextInput
        style={styles.input}
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
        placeholderTextColor={theme.colors.input.placeholder}
        editable={!disabled}
        testID="text-input"
        {...rest}
      />

      {showClearButton && value.length > 0 && !disabled && (
        <TouchableOpacity
          onPress={handleClear}
          style={styles.clearButton}
          activeOpacity={0.7}
          testID="clear-button"
        >
          <Ionicons
            name="close-circle"
            size={20}
            color={theme.colors.input.placeholder}
          />
        </TouchableOpacity>
      )}
    </View>
  );
};

