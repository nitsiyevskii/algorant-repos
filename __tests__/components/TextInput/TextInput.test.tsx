import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { TextInput } from 'src/components/TextInput/TextInput';

describe('TextInput', () => {
  const mockOnChangeText = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with default props', () => {
    const { getByPlaceholderText } = render(
      <TextInput value="" onChangeText={mockOnChangeText} />
    );
    
    expect(getByPlaceholderText('Search...')).toBeTruthy();
  });

  it('renders with custom placeholder', () => {
    const { getByPlaceholderText } = render(
      <TextInput
        value=""
        onChangeText={mockOnChangeText}
        placeholder="Custom placeholder"
      />
    );
    
    expect(getByPlaceholderText('Custom placeholder')).toBeTruthy();
  });

  it('displays the current value', () => {
    const { getByDisplayValue } = render(
      <TextInput value="test value" onChangeText={mockOnChangeText} />
    );
    
    expect(getByDisplayValue('test value')).toBeTruthy();
  });

  it('calls onChangeText when text changes', () => {
    const { getByPlaceholderText } = render(
      <TextInput value="" onChangeText={mockOnChangeText} />
    );
    
    const input = getByPlaceholderText('Search...');
    fireEvent.changeText(input, 'new text');
    
    expect(mockOnChangeText).toHaveBeenCalledWith('new text');
  });

  it('shows clear button when value is not empty', () => {
    const { queryByTestId, rerender } = render(
      <TextInput value="" onChangeText={mockOnChangeText} />
    );
    
    rerender(<TextInput value="some text" onChangeText={mockOnChangeText} />);
    
    const clearButton = queryByTestId('clear-button');
    expect(clearButton).toBeTruthy();
  });

  it('clears text when clear button is pressed', () => {
    const { getByTestId } = render(
      <TextInput value="some text" onChangeText={mockOnChangeText} />
    );
    
    const clearButton = getByTestId('clear-button');
    fireEvent.press(clearButton);
    
    expect(mockOnChangeText).toHaveBeenCalledWith('');
  });

  it('is disabled when disabled prop is true', () => {
    const { getByTestId } = render(
      <TextInput
        value=""
        onChangeText={mockOnChangeText}
        disabled={true}
      />
    );
    
    const input = getByTestId('text-input');
    expect(input.props.editable).toBe(false);
  });

  it('matches snapshot', () => {
    const { toJSON } = render(
      <TextInput value="test" onChangeText={mockOnChangeText} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});

