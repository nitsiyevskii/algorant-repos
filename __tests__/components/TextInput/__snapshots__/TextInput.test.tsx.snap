// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput matches snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#F2F2F7",
        "borderColor": "#C7C7CC",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "paddingHorizontal": 12,
      },
      {
        "opacity": 1,
      },
    ]
  }
>
  <View
    style={
      {
        "marginRight": 8,
      }
    }
  >
    <Text
      color="#8E8E93"
      name="search"
      size={20}
      testID="icon"
    >
      search
    </Text>
  </View>
  <TextInput
    editable={true}
    onChangeText={[MockFunction]}
    placeholder="Search..."
    placeholderTextColor="#8E8E93"
    style={
      {
        "color": "#000000",
        "flex": 1,
        "fontSize": 16,
        "height": "100%",
        "paddingVertical": 0,
      }
    }
    testID="text-input"
    value="test"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginLeft": 8,
        "opacity": 1,
        "padding": 4,
      }
    }
    testID="clear-button"
  >
    <Text
      color="#8E8E93"
      name="close-circle"
      size={20}
      testID="icon"
    >
      close-circle
    </Text>
  </View>
</View>
`;
